streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -176,9 +170,10 @@ public String queryableStoreName() {
 
     @Override
     public KTable<K, V> filter(final Predicate<? super K, ? super V> predicate,
-                               final Materialized<K, V, KeyValueStore<Bytes, byte[]>> materialized) {
+                               final Materialized<K, V, KeyValueStore<Bytes, byte[]>> materialized) throws UnsupportedDataTypeException {
```

Comment:
This does still enforce passing in a `KeyValueStore` -- hence, if the KTable is windowed, `validateStateStoreMatchingDataType()` would fail but there is no alternative to pass in a `WindowStore`.

It would be a "fix" to avoid unbounded growth of the result table, but still does not allow to materialize a windowed-KTable filter result.